### PR TITLE
NewStimulus: FancyBar

### PR DESCRIPTION
**Pull request by olakiril**

### CODE REVIEW

**Comment by jakereimer** [comment]
Tell us a little about FancyBar.

File: matlab/schemas/+vis/FancyBar.m [hunk 1]
```diff
@@ -0,0 +1,20 @@
+ %{
+vis.FancyBar (manual) # pink noise with periods of motion and orientation$
```
Comment (dimitri-yatsenko) [inline matlab/schemas/+vis/FancyBar.m]: The dollar sign $ at the end is a bug from a really old version of MySQL.

The comment seems inaccurate.

File: matlab/schemas/+vis/FancyBar.m [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+-> vis.Condition
+---
+pre_blank                   : double                        # (s) blank period preceding trials
+luminance                   : float                         # (cd/m^2)
+contrast                    : float                         # michelson contrast
+bar_width                   : float                         # Degrees
+grid_width                  : float                         # Degrees
+bar_speed                   : float                         # Bar speed in �/s
+flash_speed                 : float                         # cycles/sec temporal frequency of the grid flickering
+grating                     : float                         # selection beween grating (1) or flashing (0)
```
Comment (dimitri-yatsenko) [inline matlab/schemas/+vis/FancyBar.m]: what's flashing?

File: matlab/schemas/+vis/FancyBar.m [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+grat_width                  : float                         # in cycles/deg
+grat_speed                  : float                         # in cycles/deg
```
Comment (dimitri-yatsenko) [inline matlab/schemas/+vis/FancyBar.m]: the word speed implies a relationship with time.  So perhaps its cycles/sec or Hz?

File: matlab/schemas/+vis/FancyBar.m [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+axis                        : float                         # the axis of the bar movement
```
Comment (dimitri-yatsenko) [inline matlab/schemas/+vis/FancyBar.m]: perhaps direction rather than axis?

**Comment by olakiril** [comment]
Updated definition

File: matlab/schemas/+vis/FancyBar.m [hunk 5]
```diff
@@ -1,18 +1,20 @@
...
 bar_width                   : float                         # Degrees
 grid_width                  : float                         # Degrees
-bar_speed                   : float                         # Bar speed in �/s
+bar_speed                   : float                         # Bar speed in deg/s
 flash_speed                 : float                         # cycles/sec temporal frequency of the grid flickering
 grating                     : float                         # selection beween grating (1) or flashing (0)
 grat_width                  : float                         # in cycles/deg
 grat_speed                  : float                         # in cycles/deg
-axis                        : float                         # the axis of the bar movement
+axis                        : float                         # the direction of bar movement
```
Comment (dimitri-yatsenko) [inline matlab/schemas/+vis/FancyBar.m]: what are the units for axis?  degrees relative to up, clockwise?  up=0, right=90, down=180, left=270?

File: matlab/schemas/+vis/FancyBar.m [hunk 6]
```diff
@@ -1,18 +1,20 @@
  %{
-vis.FancyBar (manual) # pink noise with periods of motion and orientation$
+vis.FancyBar (manual) # Moving Bar stimulus that keeps the size and speed
```
Comment (dimitri-yatsenko) [inline matlab/schemas/+vis/FancyBar.m]: what does it mean to keep the size and speed?

Comment (olakiril) [inline matlab/schemas/+vis/FancyBar.m]: compensates for the flat screen. Things on the periphery and center do not have the same size/speed unless corrected.

File: matlab/schemas/+vis/FancyBar.m (hunk 5 above)
Comment (olakiril) [inline matlab/schemas/+vis/FancyBar.m]: Check newer version

File: matlab/schemas/+vis/FancyBar.m (hunk 6 above)
Comment (dimitri-yatsenko) [inline matlab/schemas/+vis/FancyBar.m]: okay, then let's add "in spherical coordinates centered on the subject's eye" to make it clear.

File: matlab/schemas/+vis/FancyBar.m (hunk 5 above)
Comment (dimitri-yatsenko) [inline matlab/schemas/+vis/FancyBar.m]: newer version of what?

File: matlab/schemas/+vis/FancyBar.m [hunk 7]
```diff
@@ -0,0 +1,22 @@
+ %{
+vis.FancyBar (manual) # Moving Bar stimulus that keeps the size and speed
+# of the bar constant relative to the mouse�s perspective.
```
Comment (dimitri-yatsenko) [inline matlab/schemas/+vis/FancyBar.m]: the comment must fit on one line (above) to be saved in the database.  These comments are not saved.

File: matlab/schemas/+vis/FancyBar.m [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+luminance                   : float                         # (cd/m^2)
+contrast                    : float                         # michelson contrast
+bar_width                   : float                         # Degrees
+grid_width                  : float                         # Degrees
+bar_speed                   : float                         # Bar speed in deg/s
+flash_speed                 : float                         # cycles/sec temporal frequency of the grid flickering
+grating                     : float                         # selection beween grating (1) or flashing (0)
+grat_width                  : float                         # in cycles/deg
+grat_speed                  : float                         # in cycles/deg
+axis                        : float                         # the direction of bar movement 1 --> horizontal 2-->vertical
```
Comment (dimitri-yatsenko) [inline matlab/schemas/+vis/FancyBar.m]: When the speed is positive, is the vertical movement up or down?

Comment (dimitri-yatsenko) [inline matlab/schemas/+vis/FancyBar.m]: why not use `enum('vertical', 'horizontal')` to make it self explanatory?

**Pull request merged**